File: Baekjoon/Problem_9000~9999/Problem_9842/main.py
import sys
from math import sqrt, ceil

input = lambda: sys.stdin.readline().rstrip()

def is_prime(val: int, primes: list) -> bool:
    if val == 2: return True
    
    for i in primes:
        if i > ceil(sqrt(val)): break
        if val % i == 0: return False
        
    return True

if __name__ == "__main__":
    N = int(input())
    
    val = 1
    primes = [ ]
    while len(primes) < N:
        val += 1
        if is_prime(val, primes):
            primes.append(val)

    print(primes[N-1])
    

File: Baekjoon/Problem_9000~9999/Problem_9842/main.cpp
#include <cstdio>
#include <cmath>
#include <vector>

using namespace std;

bool is_prime(const int val, const vector<int> &primes) {
    if (val == 2) return true;

    for (auto i: primes) {
        if (i > static_cast<int>(ceil(sqrt(val)))) break;
        else if (val % i == 0) return false;
    }
    return true;
}

int main() {
    int N; scanf("%d", &N);
    vector<int> primes;
    
    int val = 1;
    while (primes.size() < N) {
        if (is_prime(++val, primes)) {
            primes.push_back(val);
        }
    }

    printf("%d\n", primes[N-1]);
    
    return 0;
}